sources/lib/api/gcp/data/__init__.py

```diff
@@ -27,10 +27,109 @@ def sql(sql_template, **kwargs):
     sql_template: the template of the SQL statement with named placeholders.
     **kwargs: the dictionary of name/value pairs to use for placeholder values.
   Returns:
-    The formatted SQL statement with placeholders replaced with their values, and
-    an array of Javascript UDFs referenced by the code.
+    The formatted SQL statement with placeholders replaced with their values.
   Raises:
     Exception if a placeholder was found in the SQL statement, but did not have
     a corresponding argument value.
   """
   return _util.Sql.format(sql_template, kwargs)
+
+
+def tokenize(sql):
+  """ This is a crude tokenizer for our limited purposes.
```

Comment:
Was this just for extracting out UDF uses? Maybe you mentioned, and I don't remember, but given a list of udf names, can we just look for "name(...)"?

So rather than finding all function calls, we have a fixed list of function calls to find.